rfcs/41-ui-navigation.md

```diff
@@ -0,0 +1,364 @@
+  - `Move(Direction)`: directional gamepad or keyboard input
+  - `ScopeMove(ScopeDirection)`: tab menu hotkey
+- `NavEvent` event
+  - `InitiallyFocused`: We went from 0 focused to 1 focused element
+  - `NoChanges`: The `NavRequest` did not change the focus state.
+  - `FocusChanged`: The `NavRequest` changed the focus state,
+    has the list of items that were made inactive
+    and items made active.
+  - `Locked`: The `NavRequest` caused the focus system to lock.
+  - `Unlocked`: Something unlocked the focus system.
+- `MoveParam` trait
+  - Used to pick a focusable in a provided direction
+  - The `NavigationPlugin` is generic over the `MoveParam`
+  - `DefaultNavigationPlugins` provides a default implementation of `MoveParam`
+
+### `Focusable`
+
+A `Focusable` is an entity that can be focused.
+For gamepad-based navigation,
+this requires a way to pick a focusable in a provided direction.
+This is delegated to the UI library implementation
+through the `MoveParam` trait.
+In the case of bevy, `bevy_ui` is the UI library implementation,
+and the `GlobalTransform` is used to locate focusables in regard to each-other.
+
+The `Focusable` component holds state about what happens when it is activated
+and what focus state it is in. (focused, active, inactive, etc.)
+
+#### Note on hovering state
+
+Since navigation is completely decoupled from input and ui lib,
+it is impossible for it to tell whether a focusable is hovered.
+To keep the hovering state functionality,
+it will be necessary to add it as an independent component
+separate from the navigation library.
+
+### Input customization
+
+`ui-navigation` can be added to the app in two ways:
+1. With `NavigationPlugin`
+2. With `DefaultNavigationPlugins`
+
+(1) only inserts the navigation systems and resources,
+while (2) also adds the input systems generating the `NavRequest`s
+for gamepads, mouse and keyboard.
+
+This enables convenient defaults
+while letting users insert their own custom input logic.
+
+
+### What to focus first?
+
+Gamepad input assumes an initially focused element
+for navigating "move in direction" or "press action".
+At the very beginning of execution, there are no focused element,
+so we need fallbacks.
+By default, it is any `Focusable` when none are focused yet.
+But the user can spawn a `Focusable` as `dormant`
+and the algorithm will prefer it when no focused nodes exist yet.
+
+```rust
+self.focusables
+    .iter()
+    // The focused focusable.
+    .find_map(|(e, focus)| (focus.state == Focused).then(|| e))
+    // Dormant focusable within the root menu (if it exists)
+    .or_else(root_dormant)
+    // Any dormant focusable
+    .or_else(any_dormant)
+    // Any focusable
+    .or_else(fallback)
+```
+
+
+### Menu navigation handling
+
+The public API has `NavMenu`, but we use internally `TreeMenu`,
+this prevents end users from breaking assumptions about the menu trees.
+
+We define the `TreeMenu` component.
```

Comment:
This should try to make clear that it ties into the existing `Parent` and `Child` components.